matter/wifi/rs911x/hal/efx_spi.c

```diff
@@ -47,117 +49,24 @@
-  init.advancedSettings = &adv;   // Advanced settings structure
+#include "sl_spidrv_exp_config.h"
+extern SPIDRV_Handle_t sl_spidrv_exp_handle;
 #endif
-#if defined(EFR32MG12)
-  CMU_ClockEnable(MY_USART_CLOCK, true);
-  USART_InitSync(MY_USART, &config);
-#endif
-#if defined(EFR32MG12)
-  MY_USART->CTRL |= (1u << _USART_CTRL_SMSDELAY_SHIFT);
-  MY_USART->ROUTELOC0 =
-    (USART_ROUTELOC0_CLKLOC_LOC1) | // US2_CLK n location 1 = PA8 per datasheet section 6.4 = EXP Header pin 8
-    (USART_ROUTELOC0_CSLOC_LOC1) |  // US2_CS on location 1 = PA9 per datasheet section 6.4 = EXP Header pin 10
-    (USART_ROUTELOC0_TXLOC_LOC1) |  // US2_TX (MOSI) on location 1 = PA6 per datasheet section 6.4 = EXP Header pin 4
-    (USART_ROUTELOC0_RXLOC_LOC1);   // US2_RX (MISO) on location 1 = PA7 per datasheet section 6.4 = EXP Header pin 6
-
-  MY_USART->ROUTEPEN = USART_ROUTEPEN_TXPEN | USART_ROUTEPEN_RXPEN | USART_ROUTEPEN_CLKPEN | USART_ROUTEPEN_CSPEN;
-  MY_USART->CMD      = USART_CMD_CLEARRX | USART_CMD_CLEARTX;
-  USART_Enable(MY_USART, usartEnable);
-  GPIO_PinModeSet(SL_WFX_HOST_PINOUT_SPI_TX_PORT, SL_WFX_HOST_PINOUT_SPI_TX_PIN, gpioModePushPull, PINOUT_SET);
-  GPIO_PinModeSet(SL_WFX_HOST_PINOUT_SPI_RX_PORT, SL_WFX_HOST_PINOUT_SPI_RX_PIN, gpioModeInput, PINOUT_SET);
-  GPIO_PinModeSet(SL_WFX_HOST_PINOUT_SPI_CLK_PORT, SL_WFX_HOST_PINOUT_SPI_CLK_PIN, gpioModePushPull, PINOUT_CLEAR);
-
-#elif defined(EFR32MG21) || defined(EFR32MG24)
-  /**********************************************************************
-   * Route EUSART1 MOSI, MISO, and SCLK to the specified pins.  CS is
-   * not controlled by EUSART1 so there is no write to the corresponding
-   * EUSARTROUTE register to do this.
-   ****************************************************************************/
-  GPIO->EUSARTROUTE[1].TXROUTE = (EUS1MOSI_PORT << _GPIO_EUSART_TXROUTE_PORT_SHIFT)
-      | (EUS1MOSI_PIN << _GPIO_EUSART_TXROUTE_PIN_SHIFT);
-  GPIO->EUSARTROUTE[1].RXROUTE = (EUS1MISO_PORT << _GPIO_EUSART_RXROUTE_PORT_SHIFT)
-      | (EUS1MISO_PIN << _GPIO_EUSART_RXROUTE_PIN_SHIFT);
-  GPIO->EUSARTROUTE[1].SCLKROUTE = (EUS1SCLK_PORT << _GPIO_EUSART_SCLKROUTE_PORT_SHIFT)
-      | (EUS1SCLK_PIN << _GPIO_EUSART_SCLKROUTE_PIN_SHIFT);
-
-  // Enable EUSART interface pins
-  GPIO->EUSARTROUTE[1].ROUTEEN = GPIO_EUSART_ROUTEEN_RXPEN |    // MISO
-                                 GPIO_EUSART_ROUTEEN_TXPEN |    // MOSI
-                                 GPIO_EUSART_ROUTEEN_SCLKPEN;
-
-  // Configure and enable EUSART1
-  EUSART_SpiInit(EUSART1, &init);
-
-#else
-#error "EFRxx - No UART/HAL"
+
+#if defined(EFR32MG24)
+#include "sl_spidrv_eusart_exp_config.h"
+extern SPIDRV_Handle_t sl_spidrv_eusart_exp_handle;
 #endif
-}
+
+static unsigned int tx_dma_channel;
+static unsigned int rx_dma_channel;
+
+/*TODO -  FIX This - It belongs somewhere else depending on which USART is used */
+static uint32_t dummy_data; /* Used for DMA - when results don't matter */
```

Comment:
does this need to be a static variable? Couldn't it be declared in the function where it is required?

Reply:
dummy_data is used in multiple functions( receiveDMA and TransmitDMA) and it not used outside the scope of file.
So I think it is required to be used as static.

Reply:
If it's only use in certain functions why don't you just declare a local variable in that function? Why does those function needs a static variable if it's dummy date?

Reply:
We cannot make make dummy_data local variables in functions because of below reason
- We are passing the address of dummy_data to  DMADRV_MemoryPeripheral  in receiveDMA function.
-  We are passing the address of buf ( which is dummy_data when rx_buf is null ) to  DMADRV_PeripheralMemory  in transmitDMA function.

In both cases it cannot achieved if dummy_data is local because it will go out of scope in calling function.